pwnlib/tubes/process.py

```diff
@@ -708,22 +726,11 @@ def can_recv_raw(self, timeout):
         if not self.connected_raw('recv'):
             return False
 
-        try:
-            if timeout is None:
-                return select.select([self.proc.stdout], [], []) == ([self.proc.stdout], [], [])
+        if self.proc.stdout.readable and self.proc.stdout.readable:
+            return True
+        else:
+            return False
```

Comment:
How this works? Does it work with python 2 too? What system calls are issued under the hood?

I think you meant stdout and stderr, not twice stdout.

Reply:
select system call has different effect on linux than windows.